kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/IcebergWriter.java

```diff
@@ -52,20 +51,22 @@ public IcebergWriter(Table table, String tableName, IcebergSinkConfig config) {
 
   private void initNewWriter() {
     this.writer = Utilities.createTableWriter(table, tableName, config);
-    // FIXME: update this when the record converter is added
-    //  this.recordConverter = new RecordConverter(table, config);
+    this.recordConverter = new RecordConverter(table, config);
```

Comment:
Should we be doing some sort of validation on the config settings here or in the constructor?  What does it mean to configure a cdc field, but not have upsert mode enabled?  Alternatively what happens if you have upsert enabled but no cdc field?

It just feels like we should do some validation that users are setting correlated fields.

Reply:
Some validation is done in `IcebergSinkConfig.validate()`. The CDC and upsert related configs were removed.

Reply:
@bryanck - Is there any plan to add CDC support?  We used the tabular connector and looks like there can be an edge case when committing the equality deletes by the Coordinator.

A scenario which we are facing - 

Inserts and corresponding deletes are committed as the same commit(metadata having same Seq Num) to iceberg in form of Equality deletes (not posDeletes because the inserts went in a previous uncommitted commit), when there are very close insert and delete records ( example - records which are inserted and deleted by an Oracle Procedure to implement rejection in a specific business use case). This is intermittent and not easy to reproduce. We have created an issue for the same - https://github.com/tabular-io/iceberg-kafka-connect/issues/303

Reply:
The main concern from the community in adding CDC support is that relying on equality deletes can result in performance problems, unless you compact your data regularly. Once those concerns are addressed we can add it, and take into account the issue you're encountering.

Reply:
We didn't block equality deletes writing for Flink, we shouldn't block them for kafka connect users too. 

Since no in depth discussion happened on the performance issue, either we start a discussion to highlight the issues and find a solution together or we should just bring whatever supported in Tabular repo and improve it after that.

Reply:
Thanks for the positive responses @bryanck and @ajantha-bhat . Since the feature we are discussing is not in this repo yet, what could be a right place for the discussion? I think we can join the discussion in this issue - https://github.com/tabular-io/iceberg-kafka-connect/issues/303. Please share if you have other suggestions. 

Some more background on the issue and the environment:

Goal: Implement CDC pipeline from Oracle RDS to Iceberg S3

Components:
Source: Oracle RDS with logminer
Kafka Connect Source Connector: Debezium based (Using MSK Connect platform, version 2.7.1 AWS does not support higher atm)
Kafka Broker: MSK Provisioned ( 3.7.x.kraft)
Kafka Connect Sink Connector: tabular/iceberg-io/iceberg-kafka-connect based (**v0.6.15** -cannot use higher due to KC version 2.7.1 incompatibility)

Current State:
The Sink Connector is working fine for most part but failed UAT on some records not deleted.
We suspect an edge case in the Worker-Coordinator communication. Following is summary of flow - 
1. Coordinator sends COMMIT_REQUEST C1
2. C1 timed-out. Coordinator tries partial commit with no data committed. 
3. Workers handle COMMIT_REQUEST C1. Send COMMIT_RESPONSE and COMMIT_READY against C1. During this response insert data files are created.
4. Coordinator receives COMMIT_RESPONSE and COMMIT_READY against C1. Logs that no commit is currently in progress, however commit buffers would be updated(cmiiw).
5. Coordinator sends COMMIT_REQUEST C2
6. Workers handle COMMIT_REQUEST C2. Send COMMIT_RESPONSE and COMMIT_READY against C2. During this response the equality deletes(corresponding to the insert data files in C1) are created.
7. C2 timed-out. Coordinator tries partial commit with all new data committed with a single snapshot.
8. In iceberg the insert data files and the equality delete files are part of the same snapshot with same Seq Num. Due to iceberg constraint on the scope of equality deletes to be STRICTLY LESS than its Seq Num, the equality delete files are ignored.

Reply:
Hi @bryanck - Following up on my previous comment. We are able to implement a fix locally to handle the edge case arising from EqualityDeletes ignoring the related insert data files in the same commit. 

anmol/iceberg-kafka-connect#1

I have used `v0.6.15` as base because that is the max version we can use in MSK Connect due to Kafka version limitation. However, this would apply for `v0.6.19` as well.

The idea is to treat the envelope containing equality deletes as a standalone commit. I am also willing to work on the equalityDeletes for this repo and we can discuss the concerns/challenges which need to be addressed in adding the feature.